src/pages/workspaces/workspace/delete-workspace-modal.state.test.ts

```diff
@@ -0,0 +1,250 @@
+import { act, renderHook } from '@testing-library/react-hooks'
+import { Ajax } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+import { DeepPartial } from 'src/libs/type-utils/deep-partial'
+import { AzureWorkspaceInfo, GoogleWorkspaceInfo } from 'src/libs/workspace-utils'
+import { useDeleteWorkspaceState } from 'src/pages/workspaces/workspace/delete-workspace-modal.state'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+type AjaxExports = typeof import('src/libs/ajax')
+jest.mock('src/libs/ajax', (): AjaxExports => {
+  return {
+    ...jest.requireActual('src/libs/ajax'),
+    Ajax: jest.fn()
+  }
+})
+
+jest.mock('src/libs/error', () => ({
```

Comment:
for added type-safety on this (to ensure the mock is being constructed properly, I would add 
`type ErrorExports = typeof import('src/libs/error')`

and tweak
`jest.mock('src/libs/error', (): ErrorExports => ({`

which will add in some very helpful type checks on the replaced exports you are crafting here.  This is the general recommended pattern for safely mocking modules, as described in the Unit Testing Guide (TS)